app/services/foreman/renderer/scope/macros/kickstart.rb

```diff
@@ -0,0 +1,118 @@
+module Foreman
+  module Renderer
+    module Scope
+      module Macros
+        module Kickstart
+          include Foreman::Renderer::Errors
+          extend ApipieDSL::Module
+
+          apipie :class, desc: 'Macros to use within a kickstart template' do
+            name 'Kickstart'
+            sections only: %w[all]
+          end
+
+          # For more information about KS network directive: https://pykickstart.readthedocs.io/en/latest/kickstart-docs.html#network
+          apipie :method, "Returns a kickstart 'network' directive for a specific interface" do
+            required :iface, Nic::Managed, "Managed interface object to represent"
+            keyword :rhel_compatible, [true, false], "Is the OS rhel-compatible?"
```

Comment:
I'd prefer to avoid this. Because what is RHEL compatible? In other places we've defined it as "not Fedora", but the ecosystem is changing. It's some guess about how to interpret OS majors, but that's getting unreliable.

Let's set a minimum baseline and then add something for every capability instead.